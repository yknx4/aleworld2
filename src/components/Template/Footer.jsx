import React,{Component} from 'react';
import {
  Grid,
  Row,
  Col,
} from 'react-bootstrap';
import { Link } from 'react-router-dom';
import SocialList from '../social/SocialList';
import logo from '../../img/aleworld-logo.png';

const socialItems = [
  {
    icon: 'github',
    url: '//www.github.com/yknx4'
  },
  {
    icon: 'facebook',
    url: '//www.facebook.com/yknx4'
  },
  {
    icon: 'linkedin',
    url: '//www.linkedin.com/in/yknx4/'
  },
  {
    icon: 'instagram',
    url: '//www.instagram.com/yknx4/'
  }
];

export default class Footer extends Component {

    render() {
        return (
          <footer className="footer-2 bg-primary bg-pastel">
            <Grid>
              <Row>
                <Col md={6} mdOffset={3} sm={8} smOffset={2} className="text-center mb64 mb-xs-24">
                  <Link to="/">
                    <img alt="Logo" className="image-small mb16 outline-image" src={logo} />
                  </Link>
                  <p className="lead mb48 mb-xs-16">
                    All the magic is done<br />
                    somewhere in Colima, Mexico
                  </p>
                  <SocialList items={socialItems} />
                </Col>
              </Row>

              <Row className="fade-1-4">
                <Col sm={4} className="text-center-xs">
                  <span>2017 Ale World</span>
                </Col>

                <Col sm={4} xsHidden className="text-center">
                  <span>Life Style &amp; Software Development</span>
                </Col>

                <Col sm={4} xsHidden className="text-right">
                  <span>me@ale.world</span>
                </Col>
              </Row>
            </Grid>
          </footer>
        );
    }
}
