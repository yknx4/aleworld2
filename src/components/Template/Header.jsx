import React, { Component } from "react";
import {
  Nav,
  Module,
  ModuleGroup,
  ThemifyIcon,
  Menu,
  Dropdown,
  MegaMenu
} from "../layout";
import { Link } from "react-router-dom";

export default class Header extends Component {
  render() {
    return (
      <Nav>
        <Module className="left">
          <Link to="/">
            <h5 className="main-title">aleworld</h5>
          </Link>
        </Module>
        <Module className="widget-handle mobile-toggle right visible-sm visible-xs">
          <ThemifyIcon icon="menu" />
        </Module>
        <ModuleGroup className="right">
          <Module className="left">
            <Menu>
              <li>
                <Link to="/">Home</Link>
              </li>
              <Dropdown>
                <a href="//yknx4.github.io/blog">Blog</a>
                <MegaMenu>
                  <li>
                    <Link to="/blog">Blog</Link>
                  </li>
                </MegaMenu>
              </Dropdown>
            </Menu>
          </Module>
        </ModuleGroup>
      </Nav>
    );
  }
}
