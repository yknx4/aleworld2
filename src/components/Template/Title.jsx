import React,{Component} from 'react';

import {
  PageTitle,
} from '../layout';

import { NavLink } from 'react-router-dom';

import {
  Grid,
  Row,
  Col,
  Breadcrumb
} from 'react-bootstrap';

export default class Title extends Component {

    render() {
        return (
          <PageTitle className="page-title-1 bg-secondary">
            <Grid>
              <Row>
                <Col sm={12} className="text-center bl">
                  <span className="mb0 h1">ale</span>
                  <span className="mb0 h1 right-color">world</span>
                </Col>
                <Col sm={12} className="text-center bl">
                  <span className="mb0 h4">LifeThoughts, SocialCriticism, and some SoftwareDevelopment</span>
                </Col>
              </Row>
            </Grid>
            <Breadcrumb className='breadcrumb-2'>
              <li>
                <NavLink to="/beta" activeClassName="active">Home</NavLink>
              </li>
            </Breadcrumb>
          </PageTitle>
        );
    }
}
