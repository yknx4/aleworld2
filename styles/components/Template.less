@import '../colors';
@side-padding: 8%;
.nav-container {
  a h5 {
    display: inline;
  }
  .logo {
    padding-right: @side-padding;
  }
}

.right-color {
  color: @primary-color-dark !important;
}

.sm-img {
  width: 75%
}

.outline-image {
  -webkit-filter: drop-shadow(1px 1px 0 black)
                 drop-shadow(-1px -1px 0 black);
 filter: drop-shadow(1px 1px 0 black)
         drop-shadow(-1px -1px 0 black);
}

.mb0 {
  br {
    display: block; /* makes it have a width */
    content: ""; /* clears default height */
    margin-top: 1em; /* change this to whatever height you want it */
  }
}

.icon {
    color: @primary-color-light;
}

.bg-light {
  p {
    color: @primary-color-light;
  }
}

.bg-pastel:extend(.bg-primary) {
  background: lighten(@accent-color, 10);
}

.main-title {
    color: @main-color;
    font-family: 'texgyreadventorbold'; 
    font-size: 2.5rem;
    margin: auto;
    padding: 0;
}