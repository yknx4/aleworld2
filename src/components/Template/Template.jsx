import React, { Component } from "react";
import PropTypes from 'react-proptypes';

import {
  MainContainer,
} from '../layout';

import Header from './Header';
import Footer from './Footer';

export default class Template extends Component {
    render() {
      const {children} = this.props;
        return (
          <div>
            <Header />
            <MainContainer>
              {children}
              <Footer />
            </MainContainer>
          </div>
        );
    }
}

Template.propTypes = {
  children: PropTypes.node.isRequired,
};
