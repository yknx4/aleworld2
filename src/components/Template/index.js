import Template from './Template';
import Title from './Title';

export default Template;
export {
  Template,
  Title
};
