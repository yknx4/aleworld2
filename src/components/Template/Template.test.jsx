import React from 'react';
import { shallow, mount } from 'enzyme';
import Template from './Template';

describe('Template', () => {
  let shallowComponent;
  let mountedComponent;

  const props = {
    description: 'description',
    title: 'title',
    subtitle: 'subtitle',
    picture: {
      src: '//path',
      alt: 'alt'
    }
  };

  beforeEach(() => {
    const component = <Template {...props} > <p /> </Template>;
    shallowComponent = shallow(component);
    mountedComponent = mount(component);
  });

  it('should render without throwing an error', function() {
    expect(shallowComponent).toMatchSnapshot();
  });

  it('should be selectable by tag', function() {
    expect(shallowComponent.is('ImageTile')).toBe(true);
    expect(shallowComponent).toMatchSnapshot();
  });

  it('should mount in a full DOM', function() {
    expect(mountedComponent.find('ImageTile').length).toBe(1);
    expect(mountedComponent).toMatchSnapshot();
  });

});
