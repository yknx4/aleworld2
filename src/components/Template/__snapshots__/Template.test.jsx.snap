// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template should render without throwing an error 1`] = `
<div>
  <Header />
  <MainContainer>
     
    <p />
     
    <Footer />
  </MainContainer>
</div>
`;
